docs/bonding-intro.md

```diff
@@ -0,0 +1,255 @@
+wants, including adding new links to the group while it is being used for
+transmission, or removing links from the list if they are not to be further
+used.
+
+How the links are exactly utilized within the group, it depends on the group
+type. The simplest type, broadcast, utilizes all links at a time to send the
+same data.
+
+
+Lay-ground: using sockets for establishing a connection
+=======================================================
+
+Important changes
+-----------------
+
+Note important changes SRT underwent since the first version from UDT:
+
+1. Specifying family (`AF_INET/AF_INET6`) when creating a socket is no longer
+required. The existing `srt_socket` function redirects to a new
+`srt_create_socket` function that gets no arguments. The exact family is
+decided at the first call to `srt_bind` or `srt_connect` and it's extracted
+from the value of `sa_family` field of the `sockaddr` structure passed to
+this call.
+
+2. There's no distinction between transmission functions bound to message
+or file mode. E.g. all 3 functions: `srt_send`, `srt_sendmsg` and
+`srt_sendmsg2` can be used for sending data in any mode - all depends on what
+your application needs.
+
+
+Socket connection
+-----------------
+
+Let's review quickly what you do to establish a socket connection in the
+caller-listener arrangement.
+
+On the listener side, you create a listening endpoint. Starting with creating
+a socket:
+
+```
+int sock = srt_create_socket();
+```
+
+The listener needs to bind it first (note: simplified code):
+
+```
+sockaddr_in sa = CreateAddrInet("0.0.0.0:5000");
+srt_bind(sock, &sa, sizeof sa);
+srt_listen(sock, 5);
+sockaddr_in target;
+int connsock = srt_accept(sock, &target, sizeof target);
+```
+
+The caller side can use default system selected address and simply connect to
+the target:
+
+```
+int connsock = srt_create_socket();
+sockaddr_in sa = CreateAddrInet("target.address:5000");
+srt_connect(connsock, &sa, sizeof sa);
+```
+
+After the connection is established, you use the send/recv functions to
+transmit the data. In this case we'll utilize the most advanced versions,
+`srt_sendmsg2` and `srt_recvmsg2`.
+
+Sender side does:
+
+```
+SRT_MSGCTRL mc = srt_msgctrl_default;
+packetdata = GetPacketData();
+srt_sendmsg2(connsock, packetdata.data(), packetdata.size(), &mc);
+```
+
+Receiver side does:
+
+```
+SRT_MSGCTRL mc = srt_msgctrl_default;
+vector<char> packetdata(SRT_LIVE_DEF_PLSIZE);
+int size = srt_recvmsg2(connsock, packetdata.data(), packetdata.size(), &mc);
+packetdata.resize(size);
+```
+
+
+Group (bonded) connection
+=========================
+
+Except for several details, most of the API used for sockets can be used for
+groups. The groups also have the numeric identifiers, just like sockets, which
+are in the same domain as sockets, except that there's reserved one bit to
+mark that the identifier is for a group, bound to a `SRTGROUP_MASK` symbol.
```

Comment:
Except for several details, most of the API used for sockets can be used for
groups. Groups also have numeric identifiers, just like sockets, which
are in the same domain as sockets, except that there is one bit reserved to
indicate that the identifier is for a group, bound to a `SRTGROUP_MASK` symbol.